gladier/client.py

```diff
@@ -185,12 +190,14 @@ def tools(self):
         if getattr(self, '_tools', None):
             return self._tools
 
-        if not getattr(self, 'gladier_tools', None) or not isinstance(self.gladier_tools, Iterable):
-            raise gladier.exc.ConfigException(
-                '"gladier_tools" must be a defined list of Gladier Tools. '
-                'Ex: ["gladier.tools.hello_world.HelloWorld"]')
+        gtools = getattr(self, 'gladier_tools', [])
+        if not gtools or not isinstance(gtools, Iterable):
+            if not self.get_flow_definition():
+                raise gladier.exc.ConfigException(
+                    '"gladier_tools" must be a defined list of Gladier Tools. '
+                    'Ex: ["gladier.tools.hello_world.HelloWorld"]')
```

Comment:
Another slight change here removes this exception if a flow_definition is used instead.